src/modules/billing/services/transactions-service.js

```diff
@@ -78,7 +79,46 @@ const setErrorStatus = transactionId =>
     status: Transaction.statuses.error
   });
 
+const batchIsTwoPartTariff = batch => batch.isTwoPartTariff();
+const batchIsInReviewStatus = batch => batch.statusIsOneOf(Batch.BATCH_STATUS.review);
+const transactionIsCandidate = transaction => transaction.status === Transaction.statuses.candidate;
+
+const volumeUpdateErrors = [
+  'Batch type must be two part tariff',
+  'Batch must have review status',
+  'Transaction must have candidate status'
+];
+
+/**
+ * Validates batch, transaction and submitted volume
+ * Checks that:
+ * - the batch is a two part tariff
+ * - the batch is in review status
+ * - the transaction is in candidate status
+ *
+ * Throws error with corresponding message if criteria is not met
+ *
+ * @param  {Batch} batch   for the transaction
+ * @param  {Transaction} transaction   in question
+ * @param  {Integer} volume   to update transaction with
+ * @param  {Object} user   id and email of internal user making the update
+ */
+const updateTransactionVolume = async (batch, transaction, volume, user) => {
+  const flags = [batchIsTwoPartTariff(batch), batchIsInReviewStatus(batch), transactionIsCandidate(transaction)];
```

Comment:
I think since this work was reviewed originally, we've introduced some custom errors which it would be nice to throw here.
There is a BatchStatusError in src/modules/billing/lib/errors.js.
We can introduce a TransactionStatusError for the third one in src/modules/billing/lib/errors.js which also extends StateError.
 The benefit of these is that it's easier in consuming code to do, e.g `err instanceof BatchStatusError`, and it's also possible to check if they extend a certain base class, e.g. `err instanceof StateError` would also return true.